lib/constructs/dynamodb/dynamodb-table.ts

```diff
@@ -1,29 +1,62 @@
-import { Construct, CfnOutput } from "@aws-cdk/core";
+import { CfnOutput, Stack } from "@aws-cdk/core";
 import { Table, TableProps, BillingMode, AttributeType } from "@aws-cdk/aws-dynamodb";
 import { App } from "@serverless-stack/resources";
 
 export type DynamoDBTableProps = TableProps
 
-export class DynamoDBTable extends Construct {
-	constructor(scope: App, id: string, tableName: string, partitionKey: string, props?: DynamoDBTableProps) {
+interface DynamoDbTableDefinition {
+  tableName: string;
+  partitionKey: string;
+}
+
+const DYNAMO_TABLES: Array<DynamoDbTableDefinition> = [
+  { tableName: 'BusinessUser', partitionKey: 'uid' },
+  { tableName: 'Business', partitionKey: 'place_id' },
+  { tableName: 'Redirect', partitionKey: 'unique_id' },
+  { tableName: 'SubscriberUser', partitionKey: 'mobile_number' },
+  { tableName: 'PushMessage', partitionKey: 'uid' },
+
+];
+
+export class DynamoDBTable extends Stack {
+  scope: App;
+	constructor(scope: App, id: string) {
     super(scope, id);
-    
-    const app: any = this.node.root;
-
-    const table = new Table(this, "Table", {
-      tableName: tableName,
-      billingMode: BillingMode.PAY_PER_REQUEST, // Use on-demand billing mode
-      partitionKey: { name: partitionKey, type: AttributeType.STRING },
-    });
-    
-    // Output values
-    new CfnOutput(this, "TableName", {
-      value: table.tableName,
-      exportName: app.logicalPrefixedName("TableName"),
-    });
-    new CfnOutput(this, "TableArn", {
-      value: table.tableArn,
-      exportName: app.logicalPrefixedName("TableArn"),
-    });
+    this.scope = scope;
```

Comment:
I wonder if this is why your stack is deploying twice.

Reply:
are you referring to this.scope = scope or the table arn stuff? I tried commenting out the scope stuff in the constructor and just logged out a random string, but it still runs twice. It doesn't seem to give me an error anymore about already existing stacks so it looks like the actual database creation only occurs once now.

Reply:
Yes just the scope part. Good that it’s no longer running cfn twice but still I’m curious what is going on